lib/ruby_lsp/requests/support/common.rb

```diff
@@ -86,13 +86,16 @@ def self_receiver?(node)
           params(
             title: String,
             entries: T.any(T::Array[RubyIndexer::Entry], RubyIndexer::Entry),
+            max_entries: T.nilable(Integer),
           ).returns(T::Hash[Symbol, String])
         end
-        def categorized_markdown_from_index_entries(title, entries)
+        def categorized_markdown_from_index_entries(title, entries, max_entries = nil)
```

Comment:
Given that completion item resolve is only going to return documentation for a single item at a time, is it really necessary to limit the number of entries? Do you still see slowness even with the request separation?

Reply:
Unfortunately yes, but the issue is not in the LSP. It appears to be in VSCode when rendering the documentation. The completion item that I am using for testing is defined in about 3500 files.

Here is what a sample of the trace log looks like now, with no limit on the number of documentation entries returned:

```
2024-03-22 21:35:11.103 [info] (platform) [Trace - 9:35:11 PM] Sending request 'textDocument/foldingRange - (43)'.
2024-03-22 21:35:11.103 [info] (platform) [Trace - 9:35:11 PM] Sending request 'textDocument/documentSymbol - (44)'.
2024-03-22 21:35:11.103 [info] (platform) [Trace - 9:35:11 PM] Sending request 'textDocument/codeLens - (45)'.
2024-03-22 21:35:11.149 [info] (platform) [Trace - 9:35:11 PM] Sending request 'textDocument/semanticTokens/full - (46)'.
2024-03-22 21:35:11.165 [info] (platform) [Trace - 9:35:11 PM] Received response 'completionItem/resolve - (42)' in 63ms.
2024-03-22 21:35:18.032 [info] (platform) [Trace - 9:35:18 PM] Sending request 'textDocument/documentSymbol - (47)'.
2024-03-22 21:35:18.032 [info] (platform) [Trace - 9:35:18 PM] Sending request 'textDocument/inlayHint - (48)'.
2024-03-22 21:35:18.033 [info] (platform) [Trace - 9:35:18 PM] Sending request 'textDocument/documentLink - (49)'.
2024-03-22 21:35:18.033 [info] (platform) [Trace - 9:35:18 PM] Received response 'textDocument/foldingRange - (43)' in 6930ms.
2024-03-22 21:35:18.033 [info] (platform) [Trace - 9:35:18 PM] Received response 'textDocument/documentSymbol - (44)' in 6930ms.
2024-03-22 21:35:18.033 [info] (platform) [Trace - 9:35:18 PM] Received response 'textDocument/codeLens - (45)' in 6930ms.
2024-03-22 21:35:18.034 [info] (platform) [Trace - 9:35:18 PM] Received response 'textDocument/semanticTokens/full - (46)' in 6885ms.
```

Note that the `completionItem/resolve` request only takes 63ms, but from the client's perspective some of the other requests take ~7 seconds. During this time, the editor is completely locked up.

With a limit of 10, the trace looks like this:

```
2024-03-22 21:39:23.742 [info] (platform) [Trace - 9:39:23 PM] Sending request 'completionItem/resolve - (35)'.
2024-03-22 21:39:23.743 [info] (platform) [Trace - 9:39:23 PM] Received response 'completionItem/resolve - (35)' in 1ms.
2024-03-22 21:39:23.755 [info] (platform) [Trace - 9:39:23 PM] Sending request 'textDocument/documentHighlight - (36)'.
2024-03-22 21:39:23.757 [info] (platform) [Trace - 9:39:23 PM] Received response 'textDocument/documentHighlight - (36)' in 2ms.
2024-03-22 21:39:23.996 [info] (platform) [Trace - 9:39:23 PM] Sending request 'textDocument/codeAction - (37)'.
2024-03-22 21:39:23.997 [info] (platform) [Trace - 9:39:23 PM] Received response 'textDocument/codeAction - (37)' in 0ms.
2024-03-22 21:39:24.048 [info] (platform) [Trace - 9:39:24 PM] Sending request 'textDocument/foldingRange - (38)'.
2024-03-22 21:39:24.051 [info] (platform) [Trace - 9:39:24 PM] Received response 'textDocument/foldingRange - (38)' in 3ms.
```